server.js

```diff
@@ -24,12 +25,89 @@ const app = express();
 app.use(cors());
 app.use(express.json());
 
+const Book = mongoose.model('Book', {
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: Number,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+})
+
+if (process.env.RESET_DB) {
+  const resetDatabase = async () => {
+    await Book.deleteMany({});
+    booksData.forEach(item => {
+      const newBook = new Book(item)
+      newBook.save()
+    })
+  }
+  resetDatabase()
+}
+
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+res.send("This is an api for books.");
+});
+
+app.get('/books', async (req, res) => {
+  const Books = await Book.find({});
+  res.status(200).json ({
+    succes: true,
+    body: Books
+  })
+})
+
+app.get('/books/id/:id', async (req, res) => {
+  try {
+    const book = await Book.findById(req.params.id)
+    if (book) {
+      res.status(200).json ({
+        success: true,
+        body: book
+      });
+    } else {
+      res.status(404).json ({
+        success: false,
+        body: {
+          message: "No book with that ID was found"
+        }
+      });
+    }
+  } catch (error) {
+    res.status(400).json ({
+      success: false,
+      body: {
+        message: "Invalid Id"
+      }
+    });
+  }
 });
 
+app.get('/books/title/:title', (req, res) => {
+  Book.findOne({ title: req.params.title }).then(book => {
+    if (book) {
+      res.json(book)
+    } else {
+      res.status(404).json({ error: 'There is no book with that title.' })
+    }
+  })
+ })
+
+app.get('/books/authors/:authors', async (req, res) => {
+  const book = await Book.findOne({ authors: req.params.authors })
```

Comment:
Now only one book from an author is shown. To show all books use Book.find instead.